From 128312c2a9db9fe73057aad101f27b8dd92f8600 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Tue, 23 Dec 2014 17:29:41 -0500
Subject: [infiniband] srpt: Handle GID change events

Message-id: <519120f1571c838260cf4793ff9a0b14be249f83.1419347857.git.dledford@redhat.com>
Patchwork-id: 102388
O-Subject: [Patch RHEL7 1/2] IB/srpt: Handle GID change events
Bugzilla: 1174910
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

GID change events need a refresh just like LID change events and several
others.  Handle this the same as the others.

Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 2aa1cf64aa576671b3593d73a6cb0a28b90ee444)
Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/ulp/srpt/ib_srpt.c b/drivers/infiniband/ulp/srpt/ib_srpt.c
index ba2080f..4400f72 100644
--- a/drivers/infiniband/ulp/srpt/ib_srpt.c
+++ b/drivers/infiniband/ulp/srpt/ib_srpt.c
@@ -198,6 +198,7 @@ static void srpt_event_handler(struct ib_event_handler *handler,
  case IB_EVENT_PKEY_CHANGE:
  case IB_EVENT_SM_CHANGE:
  case IB_EVENT_CLIENT_REREGISTER:
+ case IB_EVENT_GID_CHANGE:
   /* Refresh port data asynchronously. */
   if (event->element.port_num <= sdev->device->phys_port_cnt) {
    sport = &sdev->port[event->element.port_num - 1];
-- 
1.7.1